evaporate.js

```diff
@@ -14,25 +14,27 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 
 /***************************************************************************************************
 *                                                                                                  *
-*  version 0.0.2                                                                                  *
-*                                                                                                  *
-*  TODO:                                                                                           *
-*       calculate MD5s and send with PUTs                                                          *
-*       post eTags to application server to allow resumability after client-side crash/restart      *
-*                                                                                                  *
+*  version 1.0.0                                                                                   *
 *                                                                                                  *
 ***************************************************************************************************/
 
 (function() {
-
-  var Evaporate = function(config){
-
-     this.supported = !((typeof(File)=='undefined') ||
-        (typeof(Blob)=='undefined') ||
-        !(!!Blob.prototype.webkitSlice || !!Blob.prototype.mozSlice || Blob.prototype.slice) ||
+   var FAR_FUTURE = new Date('2060-10-22');
+
+  var Evaporate = function (config) {
+
+     this.supported = !(
+        typeof File === 'undefined' ||
+        typeof Blob === 'undefined' ||
+        typeof (
+           Blob.prototype.webkitSlice ||
+           Blob.prototype.mozSlice ||
+           Blob.prototype.slice ||
+           (config.computeContentMd5 && Blob.prototype.readAsBinaryString) === 'undefined') ||
         config.testUnsupported);
 
-     if(!this.supported){
+     if (!this.supported) {
+        l.e('The broswer does not support the necessary features of File and Blob [webkitSlice || mozSlice || slice || readAsBinaryString]')
```

Comment:
l isn't actually defined until later down in this function, so calling it at this point you get a lovely `Uncaught TypeError: Cannot read property 'e' of undefined` in the console (plus there's a typo)